### DISCUSSION TITLE
[QA] Unable to figure out how to quit by output of pwd with only q and how to open a file using arrow keys

### DISCUSSION THREAD

**Discussion by madrigal1** [sayanarijit/xplr#601] [Q&A] [answered]
I have two questions that are hard to solve:

1) How to use q to trigger the exit with the pwd output instead of enter:
![image](https://user-images.githubusercontent.com/30615210/226187527-0253d764-ebde-4dad-aa9d-e70d3a0be83d.png)

2) how to use arrow right to open the file depending on filetypes (similar to ranger or lf)

any help guidance would be appreciated

**Comment by sayanarijit** [comment, marked as answer]
```lua
xplr.config.modes.builtin.default.key_bindings.on_key.right = {
  help = "enter or open",
  messages = {
    {
      BashExec0 = [===[
         test -d "${XPLR_FOCUS_PATH:?}" && "$XPLR" -m Enter || xdg-open "${XPLR_FOCUS_PATH:?}"
      ]===],
    },
  },
}

xplr.config.modes.builtin.default.key_bindings.on_key.q = {
  help = "quit printing pwd",
  messages = {
     "PrintPwdAndQuit",
  },
}
```

Something like that. I typed on mobile, so there could be typos.

See https://xplr.dev/en/configure-key-bindings#tutorial-adding-a-new-mode

Edit:
Once it's working, switch to BashExecSilently0 for snappy interaction.

**Reply by madrigal1** [reply]
For me BashExec0 was the better option as i wanted .txt files to open in nvim in the terminal whereas Silently opned it it a seprate application. But the comands work perfectly thanks a lot :)
